api/gateway/server.go

```diff
@@ -13,45 +12,42 @@ import (
 // Server represents an RPC server on the Node.
 // TODO @renaynay: eventually, gateway server should be able to be toggled on and off.
 type Server struct {
-	cfg Config
-
 	srv      *http.Server
 	srvMux   *mux.Router // http request multiplexer
 	listener net.Listener
 }
 
 // NewServer returns a new RPC Server.
-func NewServer(cfg Config) *Server {
+func NewServer(address string, port string) *Server {
 	srvMux := mux.NewRouter()
 	srvMux.Use(setContentType)
 
 	server := &Server{
-		cfg:    cfg,
 		srvMux: srvMux,
 	}
 	server.srv = &http.Server{
+		Addr:    address + ":" + port,
 		Handler: server,
 		// the amount of time allowed to read request headers. set to the default 2 seconds
 		ReadHeaderTimeout: 2 * time.Second,
 	}
 	return server
 }
 
-// Start starts the RPC Server, listening on the given address.
+// Start starts the gateway Server, listening on the given address.
 func (s *Server) Start(context.Context) error {
-	listenAddr := fmt.Sprintf("%s:%s", s.cfg.Address, s.cfg.Port)
-	listener, err := net.Listen("tcp", listenAddr)
+	listener, err := net.Listen("tcp", s.srv.Addr)
```

Comment:
Update the server to not use Listener and make lifecycling over HTTP Server directly

Reply:
https://github.com/celestiaorg/celestia-node/issues/1286